dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/trackedentity/DefaultTrackedEntityService.java

```diff
@@ -397,7 +397,10 @@ private Set<Enrollment> getEnrollments(
             e -> {
               Set<Event> filteredEvents =
                   e.getEvents().stream()
-                      .filter(event -> includeDeleted || !event.isDeleted())
+                      .filter(
+                          event ->
+                              trackerAccessManager.canRead(user, event, false).isEmpty()
```

Comment:
I am thinking of the performance implication here. 
Can we atleast move this as the second clause in this condition? 
Meaning swapping the conditions on either sides of &&.

Reply:
makes sense, thanks